core/src/main/se/tevej/game/model/resource/Resource.java

```diff
@@ -1,18 +1,27 @@
 package main.se.tevej.game.model.resource;
 
 public class Resource {
-    private double amount;
-    private ResourceType type;
+    private final double amount;
+    private final ResourceType type;
 
     public Resource(double amount, ResourceType type) {
         this.amount = amount;
         this.type = type;
     }
 
+    public Resource(Resource resource) {
+        this.amount = resource.getAmount();
+        this.type = resource.getType();
+    }
+
     public double getAmount() {
         return amount;
     }
 
+    public Resource setAmount(double amount){
```

Comment:
This does not follow the Command–query separation principle.

Reply:
I have made Resource Immutable